modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C

```diff
@@ -0,0 +1,242 @@
+      "Boolean to determine if the problem should be used in a linear or nonlinear solve");
+  params.addParam<bool>(
+      "non_equilibrium_treatment",
+      false,
+      "Use non-equilibrium wall treatement (faster than standard wall treatement)");
+  params.addParam<Real>("C1_eps", 1.44, "First epsilon coefficient");
+  params.addParam<Real>("C2_eps", 1.92, "Second epsilon coefficient");
+  params.addParam<Real>("C_mu", 0.09, "Coupled turbulent kinetic energy closure.");
+
+  params.set<unsigned short>("ghost_layers") = 2;
+  return params;
+}
+
+INSFVTKEDSourceSink::INSFVTKEDSourceSink(const InputParameters & params)
+  : FVElementalKernel(params),
+    _dim(_subproblem.mesh().dimension()),
+    _u_var(dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("u", 0))),
+    _v_var(params.isParamValid("v")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("v", 0))
+               : nullptr),
+    _w_var(params.isParamValid("w")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("w", 0))
+               : nullptr),
+    _k(getFunctor<ADReal>(NS::TKE)),
+    _rho(getFunctor<ADReal>(NS::density)),
+    _mu(getFunctor<ADReal>(NS::mu)),
+    _mu_t(getFunctor<ADReal>(NS::mu_t)),
+    _wall_boundary_names(getParam<std::vector<BoundaryName>>("walls")),
+    _max_mixing_length(getParam<Real>("max_mixing_length")),
+    _linearized_model(getParam<bool>("linearized_model")),
+    _non_equilibrium_treatment(getParam<bool>("non_equilibrium_treatment")),
+    _C1_eps(getParam<Real>("C1_eps")),
+    _C2_eps(getParam<Real>("C2_eps")),
+    _C_mu(getParam<Real>("C_mu"))
+{
+  if (!_u_var)
+    paramError("u", "the u velocity must be an INSFVVelocityVariable.");
+
+  if (_dim >= 2 && !_v_var)
+    paramError("v",
+               "In two or more dimensions, the v velocity must be supplied and it must be an "
+               "INSFVVelocityVariable.");
+
+  if (_dim >= 3 && !_w_var)
+    paramError("w",
+               "In three-dimensions, the w velocity must be supplied and it must be an "
+               "INSFVVelocityVariable.");
+}
+
+void
+INSFVTKEDSourceSink::initialSetup()
+{
+  NS::getWallBoundedElements(
+      _wall_boundary_names, _fe_problem, _subproblem, blockIDs(), _wall_bounded);
+  NS::getWallDistance(_wall_boundary_names, _fe_problem, _subproblem, blockIDs(), _dist);
+  NS::getElementFaceArgs(_wall_boundary_names, _fe_problem, _subproblem, blockIDs(), _face_infos);
+}
+
+ADReal
+INSFVTKEDSourceSink::computeQpResidual()
+{
+
+  ADReal residual = 0.0;
+  ADReal production = 0.0;
+  ADReal destruction = 0.0;
+  const Moose::StateArg state = determineState();
+  const auto old_state =
+      _linearized_model ? Moose::StateArg(1, Moose::SolutionIterationType::Nonlinear) : state;
+  const auto mu = _mu(makeElemArg(_current_elem), state);
+  const auto rho = _rho(makeElemArg(_current_elem), state);
+  const auto TKE = _k(makeElemArg(_current_elem), state);
+
+  if (_wall_bounded[_current_elem])
```

Comment:
```suggestion
  if (_wall_bounded.find(_current_elem) != _wall_bounded.end())
```
This will help keep the size of your container down which will help with the O(log N) search times through this container

Reply:
Thanks. Changed